apps/a11y-tests/src/axe-sarif-converter/axe-utils.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+/// <reference path="./axe-extension.ts" />
+import * as axe from 'axe-core';
+
+import { DictionaryStringTo } from './dictionary-types';
+
+export type ImageCodedAs = 'Decorative' | 'Meaningful';
+
+export function getMatchesFromRule(ruleId: string): (node: any, virtualNode: any) => boolean {
+  // @ts-ignore
```

Comment:
Please find a way to get rid of all the ts-ignores. If you can't figure out the proper typings, a cast to `any` is almost always the right way to go instead of totally disabling type checking for the line.

Reply:
I wouldn't go as far as saying it is the right way to go, but it is the *better* way to go :) Totally agree with this feedback though.

Reply:
On the other hand, all these code are apparently temporary..

Reply:
Really I'd say the right way is (in order of preference):
1. Type things correctly to begin with
2. For union types, use a [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) to verify and narrow the type
3. Cast to the actual type
4. Use any
5. Use `ts-ignore` (I think the only time I've used this in TS is when importing a library without typings and writing my own typings isn't feasible. I've used it more often in JS files with `ts-check` because the syntax for properly specifying a type in JS files can be significantly more difficult and verbose.)

Reply:
Agreed. `@ts-ignore` is too big a sledgehammer even for temp code. I'll remove them and see how much can be fixed.

Reply:
Removed all `@ts-ignore`s and resolved breaks.